src/Jackalope/Transport/Jackrabbit/Client.php

```diff
@@ -2122,4 +2141,102 @@ protected function getMimePart($name, $value, $mime_boundary)
 
         return $data;
     }
+
+    public function getPolicies($path)
+    {
+        return $this->getNode($path . '/rep:policy');
+    }
+
+    public function getSupportedPrivileges($path = null)
+    {
+        $path = $this->workspaceUriRoot . $path ?: '';
+
+        $request = $this->getRequest(Request::PROPFIND, $path);
+        $request->setBody($this->buildPropfindRequest(array('D:supported-privilege-set')));
+        $dom = $request->executeDom();
+
+        $set = $dom->getElementsByTagNameNS(self::NS_DAV, 'supported-privilege-set');
+        if ($set->length != 1) {
+            throw new RepositoryException('Unexpected answer from server: '.$dom->saveXML());
+        }
+
+        $privileges = array();
+        foreach ($set->item(0)->childNodes as $privilege) {
+            $privileges[] = $this->parsePrivileges($privilege);
+        }
+
+        return $privileges;
+    }
+
+    private function parsePrivileges(\DOMElement $node)
+    {
+        $privilege = null;
+        $children = array();
+
+        foreach ($node->childNodes as $child) {
+            switch ($child->tagName) {
+                case 'D:privilege':
+                    $privilege = $child;
+                    break;
+                case 'D:supported-privilege':
+                    $children[] = $this->parsePrivileges($child);
+                    break;
+                default:
+                    // ignore
+            }
+        }
+
+        if (!$privilege) {
+            throw new \Exception('invalid stuff'.$node->tagName);
+        }
+        $name = '{'.$privilege->firstChild->namespaceURI.'}'.$privilege->firstChild->localName;
+
+        return new Privilege($name, $children);
+    }
+
+    public function setPolicy(array $operation)
+    {
+        foreach ($operation as $op) $this->setPolicyss($op);
+    }
+
+    private function setPolicyss($operation)
+    {
+        if (!$operation->policy instanceof AccessControlList) {
+            throw new \Exception('wrong class');
+        }
+
+        $value = $operation->srcPath . '/rep:policy : {
+           jcr:primaryType : "rep:ACL"';
+
+        $id = 0;
+
+        foreach ($operation->policy->getAccessControlEntries() as $entry) {
+            $value .= ",\n" .
+                'entry' . $id++ . ' : {
+                    jcr:primaryType : "rep:grantACE",
+                    rep:principalName : "' . $entry->getPrincipal()->getName() . '",
+                    rep:privileges : [' . $this->buildPrivilegeList($entry) . ']
+                }';
+        }
+        $value .= '
+}
+';
+//var_dump($value);die;
+
+        $this->setJsopBody("\n+".$value, '');
+    }
+
+    private function buildPrivilegeList(AccessControlEntryInterface $entry)
+    {
+        $privileges = array();
+        foreach ($entry->getPrivileges() as $privilege) {
+            $privileges[] = str_replace('{http://www.jcp.org/jcr/1.0}', 'jcr:', $privilege->getName());
```

Comment:
the namespace handler would allow that. we should definitely not do like its now, there can be other namespaces than jcr. was a hack to see if its basically working.

and i am not entirely sure whether we really need to convert or whether jackrabbit would also understand the long name.